sr_camera_pose_estimation/src/camera_transform_publisher.py

```diff
@@ -1,120 +1,140 @@
+        try:
+            world_marker_trans = self.transform_buffer.lookup_transform(self.desired_camera_parent_frame,
+                                                                self.marker_static_tf_name, rospy.Time())
+        except (tf2_ros.LookupException, tf2_ros.ConnectivityException, tf2_ros.ExtrapolationException):
+            rospy.logerr("Failed to find world -> AR marker ({} -> {})transform.".format(
+                self.desired_camera_parent_frame, self.marker_static_tf_name))
+            return
+        self.transform = self.generate_world_camera_tf(lens_marker_pose, world_marker_trans)
+        self.publish_transform()
 
-    def generate_world_camera_tf(self, lens_marker_tf, world_marker_tf):
-        lens_marker_matrix = matrix_from_transform(lens_marker_tf.transform)
+    def generate_world_camera_tf(self, lens_marker_pose, world_marker_tf):
+        lens_marker_matrix = matrix_from_pose(lens_marker_pose)
         world_marker_matrix = matrix_from_transform(world_marker_tf.transform)
         world_lens_matrix = np.dot(world_marker_matrix, transformations.inverse_matrix(lens_marker_matrix))
         return transform_from_matrix(world_lens_matrix)
 
-    def publish_transform(self, transform):
+    def publish_transform(self):
         transform_stamped = TransformStamped()
         transform_stamped.header.stamp = rospy.get_rostime()
-        transform_stamped.transform = transform
+        transform_stamped.transform = self.transform
         transform_stamped.header.frame_id = self.desired_camera_parent_frame
         transform_stamped.child_frame_id = self.camera_root_frame
         self.broadcaster.sendTransform(transform_stamped)
 
+    def start_ar_track_alvar(self):
+        if self.alvar_process is None:
+            package = 'ar_track_alvar'
+            executable = self.alvar_node_type
+            args = '{} {} {} {} {} {}{} {}'.format(self.alvar_marker_size,
+                                                   self.alvar_max_new_marker_error,
+                                                   self.alvar_max_track_error,
+                                                   self.alvar_cam_image_topic,
+                                                   self.alvar_cam_info_topic,
+                                                   self.alvar_output_frame,
+                                                   ' {}'.format(self.alvar_med_filt_size) if
+                                                   (executable == 'findMarkerBundles') else '',
+                                                   self.alvar_bundle_files)
+            node = roslaunch.core.Node(package, executable, args=args)
+            launch = roslaunch.scriptapi.ROSLaunch()
+            launch.start()
+            self.alvar_process = launch.launch(node)
+
+    def stop_ar_track_alvar(self):
+        if not self.alvar_process is None:
+            self.alvar_process.stop()
+
 
 def matrix_from_transform(transform):
```

Comment:
Agreed. If we put that on the backlog, with a note that this should be updated when the separate utilities are ready?